File: baekjoon/greedy/2810/2810.java
import java.util.*;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.BufferedWriter;
import java.io.OutputStreamWriter;

public class Main{

  public static void main(String []args) throws Exception{
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    int N = Integer.parseInt(br.readLine());
    int[] seat = new int[N];

    String tmp = br.readLine();
    for(int i=0;i<N;i++){
      seat[i] = tmp.charAt(i);
    }
    
    int idx = 0, cnt = 0, flag = 0;
    while(idx<N){
      if(flag == 0){
        if(seat[idx] == 'S'){
          ++idx; ++cnt;
        }else{
          idx+=2; cnt+=2;
          flag = 1;
        }
      }else{
        if(seat[idx] == 'S'){
          ++idx; ++cnt;
        }else{
          idx+=2; ++cnt;
        }
      }
    }
    
    bw.write(cnt+"\n");
    bw.flush();
    bw.close();
    br.close();
  } 
}

File: baekjoon/greedy/2810/2810.py
import sys


N = int(sys.stdin.readline())
seat = list(sys.stdin.readline().rstrip())

cnt, idx, flag = 0, 0, 0
while idx < N:
  if flag == 0:
    if seat[idx] == 'S': 
      idx, cnt = idx+1, cnt+1
    else: 
      idx, cnt = idx+2, cnt+2
      flag = 1
  else:
    if seat[idx] == 'S': 
      idx, cnt = idx+1, cnt+1
    else: 
      idx, cnt = idx+2, cnt+1

print(cnt)

